pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -161,7 +162,9 @@ private static void validateDistinctQuery(PinotQuery pinotQuery)
         }
         List<Expression> orderByList = pinotQuery.getOrderByList();
         if (orderByList != null) {
-          List<Expression> distinctExpressions = function.getOperands();
+          List<Expression> distinctExpressions = function.getOperands().stream().map(
```

Comment:
I thought Pinot coding convention didn't allow using "stream()" due to performance concerns?

Reply:
revised the logic